pycbc/types/frequencyseries.py

```diff
@@ -472,12 +472,13 @@ def to_timeseries(self, delta_t=None):
             tmp = self
         else:
             tmp = FrequencySeries(zeros(flen, dtype=self.dtype),
-                             delta_f=self.delta_f, epoch=self.epoch)
+                             delta_f=self.delta_f, epoch=self.epoch,
+                             copy=False)
             tmp[:len(self)] = self[:]
 
         f = TimeSeries(zeros(tlen,
                            dtype=real_same_precision_as(self)),
-                           delta_t=delta_t)
+                           delta_t=delta_t, copy=False)
```

Comment:
Do these changes affect other schemes/backends running?

Reply:
This is a general optimization improvement.

In the previous version here, we run `zeros` to generate an array, and then in the `FrequencySeries` initialization we create *another* array of zeroes and copy across. There's no reason to copy here as the initial zeros array is not being stored anyway and is otherwise `free`d. So we should only assign the memory for this new array once, not twice, in all cases.

`copy=True` is also only *partially* working on Cupy arrays (in some cases it will fail).